adam-cli/src/main/scala/org/bdgenomics/adam/cli/Vcf2ADAM.scala

```diff
@@ -65,22 +65,24 @@ class Vcf2ADAM(val args: Vcf2ADAMArgs) extends BDGSparkCommand[Vcf2ADAMArgs] wit
     if (dictionary.isDefined)
       log.info("Using contig translation")
 
-    var adamVariants: RDD[VariantContext] = sc.loadVcf(args.vcfPath, sd = dictionary)
-    if (args.coalesce > 0) {
-      if (args.coalesce > adamVariants.partitions.size || args.forceShuffle) {
-        adamVariants = adamVariants.coalesce(args.coalesce, shuffle = true)
+    val variantContextRdd = sc.loadVcf(args.vcfPath, sdOpt = dictionary)
+    var variantContextsToSave = if (args.coalesce > 0) {
+      if (args.coalesce > variantContextRdd.partitions.size || args.forceShuffle) {
+        variantContextRdd.transform(_.coalesce(args.coalesce, shuffle = true))
```

Comment:
I'm liking this pattern less as time goes on.  Perhaps we should drop RDD from the VariantContextRDD class names for something more generic and remove the implicit conversion.  Or at least remove the implicit conversion.  What the user gets back from an ADAMContext load method should `has_a` RDD and not pretend to `is_a` RDD.